pkg/auth/service.go

```diff
@@ -835,43 +910,48 @@ func (s *KVAuthService) GetCredentialsForUser(ctx context.Context, username, acc
 	if _, err := s.GetUser(ctx, username); err != nil {
 		return nil, fmt.Errorf("%s: %w", username, err)
 	}
-	creadantialsKey := model.KvCredentialPath(username, accessKeyID)
+	credentialsKey := model.KVCredentialPath(username, accessKeyID)
 	m := model.CredentialData{}
-	_, err := s.store.GetMsg(ctx, model.PartitionKey, creadantialsKey, &m)
+	_, err := s.store.GetMsg(ctx, model.PartitionKey, credentialsKey, &m)
 	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = ErrNotFound
+		}
 		return nil, err
 	}
 	m.SecretAccessKey = ""
 	return model.CredentialFromProto(&m), nil
 }
 
 func (s *KVAuthService) GetCredentials(ctx context.Context, accessKeyID string) (*model.Credential, error) {
-	m := &model.UserData{}
-	itr, err := s.store.Scan(ctx, m.ProtoReflect().Type(), model.PartitionKey, model.KvUserPath(""))
-	if err != nil {
-		return nil, fmt.Errorf("sacn users users: %w", err)
-	}
-	defer itr.Close()
-
-	for itr.Next() {
-		if itr.Err() != nil {
-			return nil, itr.Err()
+	return s.cache.GetCredential(accessKeyID, func() (*model.Credential, error) {
+		m := &model.UserData{}
+		itr, err := s.store.Scan(ctx, m.ProtoReflect().Type(), model.PartitionKey, model.KVUserPath(""))
+		if err != nil {
+			return nil, fmt.Errorf("sacn users: %w", err)
 		}
-		entry := itr.Entry()
-		user, ok := entry.Value.(*model.UserData)
-		if !ok {
-			return nil, fmt.Errorf("list users: %w", err)
-		}
-		c := model.CredentialData{}
-		credentialsKey := model.KvCredentialPath(user.Username, accessKeyID)
-		_, err := s.store.GetMsg(ctx, model.PartitionKey, credentialsKey, &c)
-		if err != nil && errors.Is(err, kv.ErrNotFound) {
-			return nil, err
-		} else if !errors.Is(err, kv.ErrNotFound) {
-			return model.CredentialFromProto(&c), err
+		defer itr.Close()
+
+		for itr.Next() {
+			if itr.Err() != nil {
+				return nil, itr.Err()
+			}
+			entry := itr.Entry()
+			user, ok := entry.Value.(*model.UserData)
+			if !ok {
+				return nil, fmt.Errorf("list users: %w", err)
```

Comment:
Where's the error here?

Reply:
I mean, `err` might be nil

Reply:
Casting error, I changed the description.